week08/movie/src/slicers/themeSlicer.js

```diff
@@ -0,0 +1,19 @@
+import { createSlice } from "@reduxjs/toolkit";
+import { darkTheme, lightTheme } from "../Theme";
+
+const initialState = {
+  theme: darkTheme,
+};
+
+const themeSlice = createSlice({
+  name: "themeSlicer",
+  initialState: initialState,
+  reducers: {
+    changeTheme(state) {
+      state.theme = state.theme.mode === "dark" ? lightTheme : darkTheme;
+    },
+  },
+});
+
+export const { changeTheme } = themeSlice.actions;
+export default themeSlice.reducer;
```

Comment:
`@reduxjs/toolkit`으로 전역 상태를 관리해주셨네요!
최근 들어 redux을 만든 회사에서도 툴킷 사용을 권장하고 있기도 하고 툴킷 코드가 더 간단하고 명확하긴 합니다만,
2학기 때 해당 라이브러리를 배우는 과정이 있기도 하고
이번 과제는 redux의 흐름을 알기 위한 과제라 redux를 사용해주시면 더 좋았을 것 같습니다.

아래 순수 redux를 사용한 코드를 첨부해두었으니 참고만 해주세요! 이 부분은 따로 반영할 필요는 없습니다.

```suggestion
import { darkTheme, lightTheme } from "../Theme";

// 초기 상태 정의
const initialState = {
  theme: darkTheme,
};

// 액션 타입 정의
const CHANGE_THEME = "CHANGE_THEME";

// 액션 생성자 정의
export const changeTheme = () => ({
  type: CHANGE_THEME,
});

// 리듀서 정의
export const themeReducer = (state = initialState, action) => {
  switch (action.type) {
    case CHANGE_THEME:
      return {
        ...state,
        theme: state.theme.mode === "dark" ? lightTheme : darkTheme,
      };
    default:
      return state;
  }
};
```